kernel/idle.c

```diff
@@ -151,6 +177,18 @@ void idle(void *unused1, void *unused2, void *unused3)
 	}
 #else
 	for (;;) {
+#if defined(CONFIG_INIT_STACKS)
+		if (k_uptime_get_32() - idle_ts > K_SECONDS(5)) {
+			STACK_ANALYZE("interrupt stack", _interrupt_stack);
+			STACK_ANALYZE("sys work q stack", sys_work_q_stack);
+			STACK_ANALYZE("main stack", _main_stack);
+			STACK_ANALYZE("idle stack", _idle_stack);
```

Comment:
Why do the above four threads get special treatment? Wont they be included in the `k_thread_foread()`?

Reply:
No, its not included... that what needs to be debugged. May the code owner for the thread monitor feature needs to intervene here.

Reply:
The above four threads might be "special" to the kernel? They are in any case essential to the RTOS so it's critical that we get their info printed

Reply:
it is worth to check why they are not included in `k_treadh_foreach()`

Reply:
@cvinayak @ceolin Not sure why they weren't included when you tried this. Could it have been related to this: https://github.com/zephyrproject-rtos/zephyr/pull/23196?

I do get them now without that change.